components/usage/pkg/db/cost_center.go

```diff
@@ -263,3 +263,50 @@ func (c *CostCenterManager) ListLatestCostCentersWithBillingTimeBefore(ctx conte
 
 	return results, nil
 }
+
+func (c *CostCenterManager) ResetUsage(ctx context.Context, cc CostCenter) (CostCenter, *Usage, error) {
+	if cc.BillingStrategy != CostCenter_Other {
+		return CostCenter{}, nil, fmt.Errorf("cannot reset usage for Billing Strategy %s for Cost Center ID: %s", cc.BillingStrategy, cc.ID)
+	}
+
+	entity, _ := cc.ID.Values()
+
+	var spendingLimit int32
+	switch entity {
+	case AttributionEntity_Team:
+		spendingLimit = c.cfg.ForTeams
+	case AttributionEntity_User:
+		spendingLimit = c.cfg.ForUsers
+	default:
+		return CostCenter{}, nil, fmt.Errorf("cannot reset usage for unknown attribution entity ID: %s", cc.ID)
+	}
+
+	now := time.Now().UTC()
+
+	// Default to 1 month from now, if there's no nextBillingTime set on the record.
+	nextBillingTime := now.AddDate(0, 1, 0)
+	if cc.NextBillingTime.IsSet() {
+		nextBillingTime = cc.NextBillingTime.Time().AddDate(0, 1, 0)
+	}
+
+	// Create a synthetic Invoice Usage record, to reset usage
+	usage, err := c.ComputeInvoiceUsageRecord(ctx, cc.ID)
```

Comment:
It also needs to be stored. `upgradeToStripe` does this and is ill-named (doesn't do any stripe specific things). Would be good if you can rename and reuse or so

Reply:
Good spot. I'll rename the methods because `Compute` suggests it does more than just a create which doesn't align with the Go principle of communicating work/complexity.

Reply:
Fine with renaming. FWIW I usually use calculate or compute to indicate that a function doesn't do any side-effects such as persisting in DB. Does calculate imply persistence fir you?

Reply:
To some degree. I guess normally I'd be "looking out" for the "New" keyword which mostly signals it's just a constructor and not an action  but this depends.

In any case, I should've checked regardless but I've been short for time and wanted to put this up in case the team needs to be unblocked.